tests/cases/fourslash/pathCompletionsPackageJsonExportsWildcard1.ts

```diff
@@ -0,0 +1,50 @@
+/// <reference path="fourslash.ts" />
+
+// @module: nodenext
+
+// @Filename: /node_modules/foo/package.json
+//// {
+////   "name": "foo",
+////   "main": "dist/index.js",
+////   "module": "dist/index.mjs",
+////   "types": "dist/index.d.ts",
+////   "exports": {
+////     ".": {
+////       "types": "./dist/index.d.ts",
```

Comment:
While export maps like this can exist, they're always wrong because they mess up the format of the file ts sees in either cjs or esm format importers, so the nested condition pattern case is probably even more important to support.

Reply:
@weswigham would you mind expanding about what's wrong with those kinds of export maps and how they mess up with the format of the file ts sees? Just trying to solidify my understanding of how ts handles export maps (and make sure what we publish would work correctly)